dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/setting/AnalyticsTableSettings.java

```diff
@@ -125,4 +130,50 @@ Database getAndValidateDatabase(String value) {
 
     return database;
   }
+
+  /**
+   * Indicates whether to skip indexing of data element group set analytics table columns.
+   *
+   * @return {@link Skip#SKIP} if indexing should be skipped, {@link Skip#INCLUDE} if not.
+   */
+  public Skip skipIndexDataElementGroupSetColumns() {
+    return toSkip(config.isEnabled(ANALYTICS_TABLE_INDEX_DATA_ELEMENT_GROUP_SET));
+  }
+
+  /**
+   * Indicates whether to skip indexing of category analytics table columns.
+   *
+   * @return {@link Skip#SKIP} if indexing should be skipped, {@link Skip#INCLUDE} if not.
+   */
+  public Skip skipIndexCategoryColumns() {
+    return toSkip(config.isEnabled(ANALYTICS_TABLE_INDEX_CATEGORY));
+  }
+
+  /**
+   * Indicates whether to skip indexing of category option group set analytics table columns.
+   *
+   * @return {@link Skip#SKIP} if indexing should be skipped, {@link Skip#INCLUDE} if not.
+   */
+  public Skip skipIndexCategoryOptionGroupSetColumns() {
+    return toSkip(config.isEnabled(ANALYTICS_TABLE_INDEX_CATEGORY_OPTION_GROUP_SET));
+  }
+
+  /**
+   * Indicates whether to skip indexing of organisation unit group set analytics table columns.
+   *
+   * @return {@link Skip#SKIP} if indexing should be skipped, {@link Skip#INCLUDE} if not.
+   */
+  public Skip skipIndexOrgUnitGroupSetColumns() {
+    return toSkip(config.isEnabled(ANALYTICS_TABLE_INDEX_ORG_UNIT_GROUP_SET));
+  }
+
+  /**
+   * Converts the boolean enabled flag to a {@link Skip} value.
+   *
+   * @param enabled the boolean enabled flag.
+   * @return a {@link Skip} value.
+   */
+  Skip toSkip(boolean enabled) {
+    return enabled ? Skip.INCLUDE : Skip.SKIP;
```

Comment:
I haven't realized this Enum.
It sounds strange to include something part of a skip, ie: Skip.INCLUDE. In the same way, it looks weird to skip something twice, ie: Skip.SKIP.

Something like Skip.TRUE and Skip.FALSE could sound better...? Or Skip.YES, Skip.NO. These options are closer to the boolean flag `enabled` param on this method (which can be true or false).

Just wondering...

Reply:
Yes we can change to Skip.TRUE and Skip.FALSE in a separate PR. The goal is just to avoid boolean parameters in general, because it is hard to read, and error prone, as if you have many boolean parameters, you can inadvertently replace them and get different behavior.